### DISCUSSION TITLE
Problems with floating point numbers

### DISCUSSION THREAD

**Discussion by marianperca** [livewire/livewire#7309] [1. Help]
Hello! I have an issue with a component that has a floating point number property: the input value disappears when I type . (dot) or the component shows an error if I try to use that property in a computed property.
For example, I have a price and a commission (input in the interface):

```php
class ProductTest extends Component
{
    public float $price = 0;
    public float $commission = 0;

    #[Computed]
    public function finalPrice()
    {
        return $this->price + $this->commission;
    }
...
```
the input is as follows:
```html
<input type="number" step="0.01" wire:model.live="commission" ...  
...
<span class="text-lg">{{ $this->finalPrice }}</span>
```
If I clear the input or type a floating point number, I get an error: `Property [$commission] not found on component`

<img width="1395" alt="image" src="https://github.com/livewire/livewire/assets/673596/096fd67e-d828-44d9-83a4-79d131b5f9f7">

**Comment by khwadj** [comment]
Any update on this issue ?

I'm stuck with a livewire v3.3.5 for now (I'm hoping to be able to upgrade in september) so I can't tell if this is solved or not.

If it's not, here's a bit more info on this issue.

1.  It happens on an input (text or number) if that value is present at mount time and is then emptied. If the value is already empty at mount time, then the issue does not happen, the property does not dissapear.

2. In my case, the form action button calls the function save(). Debugging shows that from the first line in the save() function, the property is indeed not present on $this.


<img width="717" alt="Capture d’écran 2024-08-02 à 12 17 06" src="https://github.com/user-attachments/assets/a2023b1d-24fc-48e5-b197-522a10811a0f">



My (urgent) workaround is `$this->property = $this->property ?? NULL` (or otheriwise default value if your property is not declared as `?float`

**Comment by lcallif** [comment]
What it you were to not specify the type of variable you are defining, not give it a beginning value, and specify the input field(s) as being numeric.  This is the approach I am currently using and have not encountered an issue (yet).

It would still be a good idea to check that the variables are not Null prior to any calculations....
